installer/x86_64/install.sh

```diff
@@ -494,21 +530,38 @@ menuentry '$demo_grub_entry' {
         if [ x$grub_platform = xxen ]; then insmod xzio; insmod lzopio; fi
         insmod part_msdos
         insmod ext2
-        linux   /boot/vmlinuz-3.16.0-4-amd64 root=$demo_dev rw $GRUB_CMDLINE_LINUX  \
-                loop=$FILESYSTEM_SQUASHFS loopfstype=squashfs                       \
+        linux   /$sonic_dir/boot/vmlinuz-3.16.0-4-amd64 root=$demo_dev rw $GRUB_CMDLINE_LINUX  \
+                loop=$sonic_dir/$FILESYSTEM_SQUASHFS loopfstype=squashfs                       \
                 apparmor=1 security=apparmor $ONIE_PLATFORM_EXTRA_CMDLINE_LINUX
         echo    'Loading $demo_volume_revision_label $demo_type initial ramdisk ...'
-        initrd  /boot/initrd.img-3.16.0-4-amd64
+        initrd  /$sonic_dir/boot/initrd.img-3.16.0-4-amd64
 }
 EOF
 
-# Add menu entries for ONIE -- use the grub fragment provided by the
-# ONIE distribution.
-$onie_root_dir/grub.d/50_onie_grub >> $grub_cfg
+if [ "$environment" != "sonic" ]; then
+    # Add menu entries for ONIE -- use the grub fragment provided by the
+    # ONIE distribution.
+    $onie_root_dir/grub.d/50_onie_grub >> $grub_cfg
+    mkdir -p $onie_initrd_tmp/$demo_mnt/grub
+else
+cat <<EOF >> $grub_cfg
+$old_grub_menuentry
+$grub_menuentry
+EOF
+fi
 
-mkdir -p $onie_initrd_tmp/$demo_mnt/grub
 cp $grub_cfg $onie_initrd_tmp/$demo_mnt/grub/grub.cfg
 
+# Remove extra SONiC image if any
+if [ "$environment" = "sonic" ]; then
```

Comment:
Enumerating thru the directories is much simpler than thru the menuitems in grub.cfg
Also in this case we get the exact image directory name we need to remove.